platform-hub-api/app/models/user.rb

```diff
@@ -69,4 +80,17 @@ def update_flag! name, value
     self.save!
   end
 
+  def touch_and_update_main_identity payload
+    self.with_lock do
+      self.touch :last_seen_at
+    end
+
+    i = main_identity
+    i.with_lock do
+      if i.present? && i.external_id != payload['sub']
```

Comment:
`i.present?` check probably needs to go before the `i.with_local` (otherwise there may be a null pointer error, I think).